### ISSUE TITLE
Actually when we run Script it gives Native app, how can we do same thing for Web App/Server-side app

### ISSUE THREAD

**Issue by krishnareddyg** [CriticalPathTraining/PowerBiEmbedded#4]
Hi Ted,

Actually it create Native app, could you please suggest me how to do the Web app for same thing, Thanks.

**Comment by TedPattison** [comment]
Yes, my utility app creates a "native app" in Azure and not a "web app" because web app registration is not currently supported in Power BI embedded. The only supported way to use 3rd party Power BI embedding 
today is to create the app as a native app. I certainly hope that the Power BI embedding plumbing evolves to support registering a Power BI embedding app as a web app before too long because it will make things far more secure., and when it does I will update this code.

FWIW, it is the code that sets the application's **PublicClient** property to **true** that makes the app a "native app" in Azure as opposed to a "web app".

**Comment by krishnareddyg** [comment]
Hi Ted,

Thanks for your response, I have one more question regarding Pbix installer for PowerBI script.

It actually pull the pbix files from local server to power bi workspace and also it update Azure sql credentials.

Is there any possibility we can do same thing for Web app instead of Native app.
So that we can create Web app manually in Azure AD and call Client ID and Secret key in the script, right now it will accept only Native app and only it takes client ID, if you know what needs to be change please suggest me, Thanks.

Regards,
Krishna
(732)524-4230

**Comment by TedPattison** [comment]
Hi Krishna,
You only need to install as a "native app" for one specific scenario when you are using 3rd party Power BI embedding because you must authenticate the master user account without an interactive behavior. However, for general usage of the Power BI Service API, you can certainly register your application as a "Web App" and use the Authorization Code Grant flow to authenticate the user in an interactive session. If you use Authorization Code Grant flow, you will need to register a key for the client secret and use that key as part of the authentication process.
